core/files/uploads-manager.php

```diff
@@ -271,7 +271,16 @@ public function remove_file_or_dir( $path ) {
 	 * @return string|\WP_Error
 	 */
 	public function create_temp_file( $file_content, $file_name ) {
-		$temp_filename = $this->create_unique_dir() . $file_name;
+		/**
+		 * Temp File Path
+		 *
+		 * Allows modifying the full path of the temporary file.
+		 *
+		 * @since 3.7.0
+		 *
+		 * @param string full path to file
+		 */
+		$temp_filename = apply_filters( 'elementor/files/temp-file-path', $this->create_unique_dir() . $file_name );
```

Comment:
```suggestion
		$temp_filename = $this->create_unique_dir() . $file_name;

		/**
		 * Temp File Path
		 *
		 * Allows modifying the full path of the temporary file.
		 *
		 * @since 3.7.0
		 *
		 * @param string full path to file
		 */
		$temp_filename = apply_filters( 'elementor/files/temp-file-path', $temp_filename );
```